feat: add `autogram.Engine` (batched) (#387)
TorchJD/torchjd: docs/source/index.rst

Gradient descent relies on gradients to optimize a single objective. Jacobian descent takes this
idea a step further, using the Jacobian to optimize multiple objectives. An important component of
Jacobian descent is the aggregator, which maps the Jacobian to an optimization step. In the page

> :doc:`Aggregation <docs/aggregation/index>`, we provide an overview of the various aggregators
> available in TorchJD, along with some of their key characteristics. A precise description of this
> formalism along with the :doc:`UPGrad <docs/aggregation/upgrad>` aggregator is available in Section

Author: ```suggestion
:doc:`Aggregation <docs/aggregation/index>`, we provide an overview of the various aggregators
available in TorchJD, and their corresponding weightings.
```

We don't give the "key characteristics" anymore, and emphasizing UPGrad here is a bit useless IMO (it's already the recommended aggregator).
